### ISSUE TITLE
How can Prebid.js and GAM/GPT function together in Fledge?

### ISSUE THREAD

**Issue by JoelPM** [google/ads-privacy#57]
As noted in the WICG Fledge call today, we are trying to understand how Prebid.js will function in Fledge. The biggest unknown is how the publisher's ad server (most often GAM) intends to function in Fledge. In the present world, the final ad selection happens inside of GAM using proxy line items to represent Prebid.js demand. In Fledge, the browser will run the final ad selection process and this necessitates the consolidation of all demand in the Fledge Auction. This is made somewhat easier by the introduction of Component Auctions, which allow each partner to run their own Interest Group auction. However, several challenges remain to be solved regarding the uber auction. Those are:

1. Who owns and configures the uber auction?
2. Inputs are necessary for the uber auction

## Who owns and configures the uber auction?
The uber auction code needs to be publisher configurable (to meet their business use cases) while also being a standard that different partners can integrate with. There are compelling arguments for making this code open-source and consortium owned (explored in brief [here](https://github.com/JoelPM/prebid-td/blob/main/PrebidBackground.md#why-prebidjs-works)). Prebid.js has evolved as a solution that allows competitors to work together in a standardized way on the publisher's behalf and this seems like a good model to continue.

## Inputs necessary for the uber auction
The uber auction requires several things:
1) AuctionConfig for every component auction.
2) The maximum contextual bid price (and potentially priority).

The need for component auction `AuctionConfig`s is self explanatory. The maximum contextual bid price is necessary to allow the uber auction to determine if any of the component auctions performed better than the contextual ad selection process.

# Needed from GAM/GPT
## `AuctionConfig`
If GPT were to provide a mechanism by which an `AuctionConfig` object could be retrieved, it could be added as a component auction of the uber auction for the ad slot.

## Bid/Price Signal
All the Prebid.js SSPs currently provide a price signal that is used by Prebid.js to determine the contextual winner. Something similar would be needed from GPT in order to determine what the price to beat is for the Fledge auction.

In short, it would be great to work with the GAM team to devise a solution that enables publishers to easily integrate Fledge demand in the future.

cc @jeffkaufman @sbelov @brodrigu @jonasz @fbastello

*Further Reading*
An overview of our current thinking can be found [here](https://github.com/JoelPM/prebid-td/blob/main/PrebidFledge.md). A partial sequence diagram that prompted this issue:

![Prebid/Fledge sequence diagram](https://raw.githubusercontent.com/JoelPM/prebid-td/main/out/prebid_fledge_overview/prebid_fledge_overview.png)

**Comment by jeffkaufman** [comment]
Hi Joel, I just wanted to quickly reply to let you know that we're thinking about how we could support multi-seller FLEDGE, and we'll get back to you once we have a better idea what that might look like.

**Comment by onetag-dev** [comment]
Since some time has passed, and in the meantime Prebid.js has released the integration module for FLEDGE (fledgeForGpt), it's not yet clear how the top-level seller (in this case, GPT) compares contextual bids with those derived from IG signals. In cases of FLEDGE eligibility, adapters can return a tuple consisting of contextual bids and FLEDGE Auction Configs inside the interpretResponse function. As also described in this issue, we should expect that the top-level runner can choose the winning bid between the 'classic' contextual auction and the FLEDGE one.

Is this assumption correct? If so, how is the comparison performed? Does GPT compare the bid value of the contextual winner with that of the FLEDGE auction? It would be great to have some documentation references, let us know if it would be more appropriate to raise this request in a new issue, thank you
